pallets/nfts/src/lib.rs

```diff
@@ -1295,80 +1340,128 @@ pub mod pallet {
-		/// - `Signed` with the signer being the Owner of the `item`;
+		/// - `Signed` with the signer being the Owner of the `item` if the `item` is provided;
 		///
 		/// Arguments:
 		/// - `collection`: The collection of the item of whose approval will be cancelled.
-		/// - `item`: The item of the collection of whose approval will be cancelled.
+		/// - `maybe_item`: The optional item of the collection of whose approval will be cancelled.
+		///   If not provided, approval to transfer items in the collection that owned by `origin`
+		///   will be cancelled.
 		/// - `delegate`: The account that is going to loose their approval.
 		///
 		/// Emits `ApprovalCancelled` on success.
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(16)]
-		#[pallet::weight(T::WeightInfo::cancel_approval())]
+		#[pallet::weight(T::WeightInfo::cancel_approval(maybe_item.is_some() as u32))]
 		pub fn cancel_approval(
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
-			item: T::ItemId,
+			maybe_item: Option<T::ItemId>,
 			delegate: AccountIdLookupOf<T>,
 		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
 			let delegate = T::Lookup::lookup(delegate)?;
-			Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+			match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
+					Self::do_cancel_collection_approval(origin, collection, delegate)
+				},
+			}
 		}
 
 		/// Cancel all the approvals of a specific item.
 		///
 		/// Origin must be either:
 		/// - the `Force` origin;
-		/// - `Signed` with the signer being the Owner of the `item`;
+		/// - `Signed` with the signer being the Owner of the `item` if the `item` is provided;
 		///
 		/// Arguments:
 		/// - `collection`: The collection of the item of whose approvals will be cleared.
-		/// - `item`: The item of the collection of whose approvals will be cleared.
+		/// - `maybe_item`: The item of the collection of whose approvals will be cleared. The
+		///   optional item of the collection of whose approval will be cleared. If not provided,
+		///   all approvals to transfer items within collection.
+		/// - `witness_approvals`: Information on the collection approvals cleared. This must be
+		///   correct.
 		///
 		/// Emits `AllApprovalsCancelled` on success.
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(17)]
-		#[pallet::weight(T::WeightInfo::clear_all_transfer_approvals())]
+		#[pallet::weight(
+		    T::WeightInfo::clear_all_transfer_approvals(
+				maybe_item.is_some() as u32,
+				T::ApprovalsLimit::get()
```

Comment:
Should this not be the witness approvals value instead, defaulting to T::ApprovalsLimit for items? As below, I think we should consider removing the option type to make the simpler to manage.